src/mon/MonClient.cc

```diff
@@ -164,13 +172,15 @@ int MonClient::get_monmap_and_config()
 	r = 0;
 	break;
       }
-      while ((!got_config || monmap.get_epoch() == 0) && r == 0) {
+      while ((!bootstrap_config || monmap.get_epoch() == 0) && r == 0) {
 	ldout(cct,20) << __func__ << " waiting for monmap|config" << dendl;
 	map_cond.wait_for(l, ceph::make_timespan(
           cct->_conf->mon_client_hunt_interval));
       }
-      if (got_config) {
+
+      if (bootstrap_config) {
 	ldout(cct,10) << __func__ << " success" << dendl;
+	config = std::move(bootstrap_config);
```

Comment:
As I understand all those refs, when `get_monmap_and_config` returns the `config` will be destroyed, destroying `MConfig` object from `handle_config`?

Just wondering if it is safe to destroy the object instead of calling `put` method, as `handle_config` does?

Reply:
When wrapped in a `ceph::ref_t`, the reference counting is handled automatically to avoid the need to `get`/`put`.

Reply:
Hm, I am sure I am missing something, but I had an impression that when wrapping `MConfig` in a `ceph::ref_t` you ended up with two ref counters -- one inherited by `MConfig` from `Message -> RefCountedObject` and one from `ref_t` (`intrusive_ptr`). And they looked to me as independent , and when `ref_t` counter was reaching 0 the pointed object was just destroyed.

Reply:
yeah, that's the normal behavior. By passing `false` as the second param in the constructor, it instructs it to not increment the counter.